di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/handlers/AuthorizeHandler.java

```diff
@@ -209,43 +212,37 @@ private void persistData(QueryParamsMap queryParamsMap, AuthorizationCode author
     }
 
     private ValidationResult validateQueryParams(QueryParamsMap queryParams) {
-        String redirectUriValue = queryParams.value(RequestParamConstants.REDIRECT_URI);
-        if (Validator.isNullBlankOrEmpty(redirectUriValue)) {
-            return new ValidationResult(false, new ErrorObject(
-                    ERROR_CODE_INVALID_REDIRECT_URI,
-                    "redirect_uri param must be provided",
-                    HttpServletResponse.SC_BAD_REQUEST));
-        }
-
         String responseTypeValue = queryParams.value(RequestParamConstants.RESPONSE_TYPE);
         if (Validator.isNullBlankOrEmpty(responseTypeValue) || !responseTypeValue.equals(ResponseType.Value.CODE.getValue())) {
             return new ValidationResult(false, OAuth2Error.UNSUPPORTED_RESPONSE_TYPE);
         }
 
         String clientIdValue = queryParams.value(RequestParamConstants.CLIENT_ID);
-        if (Validator.isNullBlankOrEmpty(clientIdValue)) {
+        if (Validator.isNullBlankOrEmpty(clientIdValue) || CredentialIssuerConfig.getClientConfig(clientIdValue) == null) {
             return new ValidationResult(false, OAuth2Error.INVALID_CLIENT);
         }
 
+        if (Validator.redirectUrlIsInvalid(queryParams)) {
+            return new ValidationResult(false, new ErrorObject(
+                    ERROR_CODE_INVALID_REDIRECT_URI,
+                    "redirect_uri param must be provided",
+                    HttpServletResponse.SC_BAD_REQUEST));
+        }
+
         return ValidationResult.createValidResult();
     }
 
     private String getSharedAttributes(QueryParamsMap queryParamsMap) {
         String requestParam = queryParamsMap.value(RequestParamConstants.REQUEST);
         String clientIdParam = queryParamsMap.value(RequestParamConstants.CLIENT_ID);
 
-        if (StringUtils.isBlank(CredentialIssuerConfig.CLIENT_CONFIG)) {
+        if (CredentialIssuerConfig.CLIENT_CONFIGS.isEmpty()) {
```

Comment:
Slightly concerned about this bit and specifically when the CLIENT_CONFIGS var gets set with a value. As it starts out as null and only gets a value after the 1st call to CredentialIssuerConfig.getClientConfig() I think? Not sure if there is a danger of a null pointer here?

Reply:
yeah, good shout. I'll update it to something null safe.